packages/requisitions/src/RequestRequisition/api/hooks.ts

```diff
@@ -202,3 +204,33 @@ export const useDeleteSelectedRequisitions = () => {
 
   return deleteAction;
 };
+
+export const useAddFromMasterList = () => {
+  const queryClient = useQueryClient();
+  const { id: requestId, requisitionNumber } = useRequestFields([
+    'id',
+    'requisitionNumber',
+  ]);
+  const api = useRequestApi();
+  const mutationState = useMutation(api.addFromMasterList, {
+    onSettled: () =>
+      queryClient.invalidateQueries(api.keys.detail(String(requisitionNumber))),
+  });
+
+  const t = useTranslation('distribution');
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('message.confirm-add-from-master-list'),
+    onConfirm: () => {},
+  });
+
+  const addFromMasterList = async ({
+    id: masterListId,
+  }: MasterListRowFragment) => {
+    getConfirmation({
+      onConfirm: () => mutationState.mutateAsync({ masterListId, requestId }),
+    });
+  };
+
+  return { ...mutationState, addFromMasterList };
+};
```

Comment:
This is *usually* quite an annoying flow to do 

- open a search modal 
- close it
- show a confirmation
- trigger ab=n action with the original selected value

pretty happy how it's turned out.. a part from the optional `onChange`